oem/supermicro/ikvm.go

```diff
@@ -0,0 +1,78 @@
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+package supermicro
+
+import (
+	"encoding/json"
+	"reflect"
+
+	"github.com/stmcginnis/gofish/common"
+)
+
+type IKVMInterface string
+
+const (
+	JavaIKVMInterface IKVMInterface = "JAVA plug-in"
+	HTMLIKVMInterface IKVMInterface = "HTML 5"
+)
+
+type IKVM struct {
+	common.Entity
+
+	OdataType        string        `json:"@odata.type"`
+	CurrentInterface IKVMInterface `json:"Current interface"`
+	URI              string        `json:"URI"`
+
+	rawData []byte
+}
+
+func (ikvm *IKVM) UnmarshalJSON(b []byte) error {
```

Comment:
It doesn't look like this or `Update` are needed since there is nothing special needed for deserialization for JSON and there are no updatable fields.

No harm having them though.

Reply:
The `CurrentInterface` is changeable and we use it to change the setting from Java to HTML. This populates the URI which can be used without further authentication: 

```go
func (barsch *Barsch) GetIKVM() error {
	ikvm, err := barsch.Manager.IKVM()
	if err != nil {
		return fmt.Errorf("IKVM: %v", err)
	}

	if ikvm.CurrentInterface != supermicro.HTMLIKVMInterface {
		ikvm.CurrentInterface = supermicro.HTMLIKVMInterface
		if err := ikvm.Update(); err != nil {
			return fmt.Errorf("can't switch to HTML5 IKVM: %v", err)
		}
		ikvm, err = barsch.Manager.IKVM()
		if err != nil {
			return fmt.Errorf("can't update IKVM after switch to HTML Interface: %v", err)
		}
	}

	log.Printf("URI=%s%s", barsch.hostname, ikvm.URI)

	return nil
}
```

(Current versions of the X12 IPMI firmware will return a 404, but it did work on older ones - I'll probably open a bug report with supermicro about it)